Source/WebCore/Modules/mediastream/ImageCapture.cpp

```diff
@@ -53,6 +54,16 @@ ImageCapture::ImageCapture(Document& document, Ref<MediaStreamTrack> track)
 
 ImageCapture::~ImageCapture() = default;
 
+void ImageCapture::takePhoto(PhotoSettings&& settings, DOMPromiseDeferred<IDLInterface<Blob>>&& promise)
+{
+    m_track->takePhoto(WTFMove(settings))->whenSettled(RunLoop::main(), [protectedThis = Ref { *this }, promise = WTFMove(promise)] (auto&& result) mutable {
+        if (!result)
+            promise.reject(WTFMove(result.error()));
+        else
+            promise.resolve(Blob::create(protectedThis->scriptExecutionContext(), WTFMove(get<0>(result.value())), WTFMove(get<1>(result.value()))));
```

Comment:
We want to resolve the promise in an event task queue, which is not guaranteed here.

Maybe we actually want to specialise SerialFunctionDispatcher for ScriptExecutionContext (or maybe Document) so that we ensure that the callback is called within an event task.
It would also write better since we would not use `RunLoop::main()` but `scriptExecutionContext ()`